parcels/fieldset.py

```diff
@@ -97,6 +97,8 @@ def check_complete(self):
             g.check_zonal_periodic()
             if g is not ugrid and g.time_origin != 0:
                 g.time = g.time + (g.time_origin - ugrid.time_origin) / np.timedelta64(1, 's')
+                if g.defer_load:
+                    g.time_full = g.time
```

Comment:
Doesn't this also need a simple unit test, by the way? So that we can catch #351 if it appears again?

Reply:
Done